<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>IncrediBuild::Internal::BuildConsoleBuildStep</class>
 <widget class="QWidget" name="IncrediBuild::Internal::BuildConsoleBuildStep">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>941</width>
    <height>867</height>
   </rect>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="22" column="1">
    <widget class="Utils::PathChooser" name="logFilePathChooser" native="true"/>
   </item>
   <item row="20" column="1">
    <widget class="Utils::PathChooser" name="monFilePathChooser" native="true"/>
   </item>
   <item row="4" column="1">
    <widget class="Utils::PathChooser" name="makePathChooser" native="true"/>
   </item>
   <item row="11" column="1">
    <widget class="Utils::PathChooser" name="profileXmlPathChooser" native="true"/>
   </item>
   <item row="28" column="0">
    <widget class="QLabel" name="label_6">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Miscellaneous</string>
     </property>
    </widget>
   </item>
   <item row="16" column="0">
    <widget class="QLabel" name="newestOsLabel">
     <property name="toolTip">
      <string>Specifies the newest operating system installed on a helper machine to be allowed to participate as helper in the build.</string>
     </property>
     <property name="text">
      <string>Newest allowed helper machine OS:</string>
     </property>
    </widget>
   </item>
   <item row="10" column="0">
    <widget class="QLabel" name="label_4">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>IncrediBuild Distribution control</string>
     </property>
    </widget>
   </item>
   <item row="17" column="0">
    <widget class="QLabel" name="oldestOsLabel">
     <property name="toolTip">
      <string>Specifies the oldest operating system installed on a helper machine to be allowed to participate as helper in the build.</string>
     </property>
     <property name="text">
      <string>Oldest allowed helper machine OS:</string>
     </property>
    </widget>
   </item>
   <item row="19" column="1">
    <widget class="QLineEdit" name="titleEdit"/>
   </item>
   <item row="19" column="0">
    <widget class="QLabel" name="titleLabel">
     <property name="toolTip">
      <string>Specifies a custom header line which will be displayed in the beginning of the build output text. This title will also be used for the Build History and Build Monitor displays.</string>
     </property>
     <property name="text">
      <string>Build Title:</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_3">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Target and configuration</string>
     </property>
    </widget>
   </item>
   <item row="22" column="0">
    <widget class="QLabel" name="logFileLabel">
     <property name="toolTip">
      <string>Writes build output to a file</string>
     </property>
     <property name="text">
      <string>Output Log file:</string>
     </property>
    </widget>
   </item>
   <item row="20" column="0">
    <widget class="QLabel" name="monFileLabel">
     <property name="toolTip">
      <string>Writes a copy of the build progress (.ib_mon) file to the specified location. - If only a folder name is given, IncrediBuild generates a GUID for the file name. - A message containing the location of the saved .ib_mon file is added to the end of the build output</string>
     </property>
     <property name="text">
      <string>Save IncrediBuild monitor file:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="makeLabel">
     <property name="text">
      <string>Make command:</string>
     </property>
    </widget>
   </item>
   <item row="16" column="1">
    <widget class="QComboBox" name="newestWindowsOs"/>
   </item>
   <item row="15" column="1">
    <widget class="QSpinBox" name="maxCpuSpin">
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="24" column="0">
    <widget class="QCheckBox" name="showAgents">
     <property name="toolTip">
      <string>Shows the Agent used to build each file</string>
     </property>
     <property name="text">
      <string>Show Agents in output</string>
     </property>
    </widget>
   </item>
   <item row="15" column="0">
    <widget class="QLabel" name="maxCpuLabel">
     <property name="toolTip">
      <string>Determines the maximum number of CPU cores that can be used in a build, regardless of the number of available Agents. It takes into account both local and remote cores, even if the Avoid Task Execution on Local Machine option is selected.</string>
     </property>
     <property name="text">
      <string>Maximum CPUs to utilize in the build:</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="makeArgumentsLabel">
     <property name="text">
      <string>Make arguments:</string>
     </property>
    </widget>
   </item>
   <item row="30" column="0">
    <widget class="QCheckBox" name="stopOnError">
     <property name="toolTip">
      <string>When specified, the execution will stop as soon as an error is encountered.  This is the default behavior in 'Visual Studio' builds, but not the default for 'Make and Build tools' or 'Dev Tools' builds</string>
     </property>
     <property name="text">
      <string>Stop On Errors</string>
     </property>
    </widget>
   </item>
   <item row="18" column="0">
    <widget class="QLabel" name="label_5">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Output and Logging</string>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <widget class="QLabel" name="profileXmlLabel">
     <property name="toolTip">
      <string>The Profile XML file is used to define how Automatic Interception Interface should handle the various processes involved in a distributed job. It is not necessary for 'Visual Studio' or 'Make and Build tools' builds,  but can be used to provide configuration options if those builds use additional processes that are not included in those packages. it is required to configure distributable processes in 'Dev Tools' builds.</string>
     </property>
     <property name="text">
      <string>Profile.xml:</string>
     </property>
    </widget>
   </item>
   <item row="17" column="1">
    <widget class="QComboBox" name="oldestWindowsOs"/>
   </item>
   <item row="12" column="0">
    <widget class="QCheckBox" name="avoidLocal">
     <property name="toolTip">
      <string>Overrides the Agent Settings dialog Avoid task execution on local machine when possible option. This allows to free more resources on the initiator machine and could be beneficial to distribution in scenarios where the initiating machine is bottlenecking the build with High CPU usage</string>
     </property>
     <property name="text">
      <string>Avoid Local</string>
     </property>
    </widget>
   </item>
   <item row="21" column="0">
    <widget class="QCheckBox" name="suppressStdOut">
     <property name="toolTip">
      <string>Does not write anything to the standard output</string>
     </property>
     <property name="text">
      <string>Suppress STDOUT</string>
     </property>
    </widget>
   </item>
   <item row="5" column="1">
    <widget class="QLineEdit" name="makeArgumentsLineEdit"/>
   </item>
   <item row="27" column="1">
    <widget class="QComboBox" name="logLevel"/>
   </item>
   <item row="32" column="0">
    <widget class="QCheckBox" name="openMonitor">
     <property name="toolTip">
      <string>Opens an IncrediBuild Build Monitor that graphically displays the buildâ€™s progress once the build starts.</string>
     </property>
     <property name="text">
      <string>Open Monitor</string>
     </property>
    </widget>
   </item>
   <item row="29" column="1">
    <widget class="QLineEdit" name="setEnvEdit"/>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label">
     <property name="toolTip">
      <string>Select an helper to establish the build command.</string>
     </property>
     <property name="text">
      <string>Command Helper:</string>
     </property>
    </widget>
   </item>
   <item row="31" column="0">
    <widget class="QLabel" name="additionalArgsLabel">
     <property name="toolTip">
      <string>Add additional buildconsole arguments manually. The value of this field will be concatenated to the final buildconsole command line</string>
     </property>
     <property name="text">
      <string>Additional Arguments:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic; color:#ff0000;&quot;&gt;Please enter the appropriate arguments to your build command.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="29" column="0">
    <widget class="QLabel" name="setEnvLabel">
     <property name="toolTip">
      <string>Sets or overrides environment variables for the context of the build</string>
     </property>
     <property name="text">
      <string>Set an Environment Variable:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QComboBox" name="commandBuilder"/>
   </item>
   <item row="8" column="0">
    <widget class="QCheckBox" name="keepJobsNum">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Setting this option to true, forces IncrediBuild to not override the -j command line switch. &lt;/p&gt;&lt;p&gt;The default IncrediBuild behavior is to set a high value to the -j command line switch which controls the number of processes that the build tools executed by QT will execute in parallel (the default IncrediBuild behavior will set this value to 200).&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Keep Original Jobs Num</string>
     </property>
    </widget>
   </item>
   <item row="31" column="1">
    <widget class="QLineEdit" name="additionalArgsEdit"/>
   </item>
   <item row="27" column="0">
    <widget class="QLabel" name="logLevelLabel">
     <property name="toolTip">
      <string>Overrides the internal Incredibuild logging level for this build. Does not affect output or any user accessible logging. Used mainly to troubleshoot issues with the help of IncrediBuild support</string>
     </property>
     <property name="text">
      <string>Internal IncrediBuild logging level:</string>
     </property>
    </widget>
   </item>
   <item row="25" column="0">
    <widget class="QCheckBox" name="showTime">
     <property name="toolTip">
      <string>Shows the Start and Finish time for each file built</string>
     </property>
     <property name="text">
      <string>Show Time in output</string>
     </property>
    </widget>
   </item>
   <item row="23" column="0">
    <widget class="QCheckBox" name="showCmd">
     <property name="toolTip">
      <string>Shows, for each file built, the command-line used by IncrediBuild to build the file</string>
     </property>
     <property name="text">
      <string>Show Commands in output</string>
     </property>
    </widget>
   </item>
   <item row="26" column="0">
    <widget class="QCheckBox" name="hideHeader">
     <property name="toolTip">
      <string>Suppresses the 'IncrediBuild' header in the build output</string>
     </property>
     <property name="text">
      <string>Hide IncrediBuild Header in output</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0" colspan="2">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic; color:#ff0000;&quot;&gt;Please make sure the build command's multi-job parameter value is large enough (such as -J200 for the JOM or Make build tools)&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
   <slots>
    <signal>editingFinished()</signal>
    <signal>browsingFinished()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
